### PR TITLE
Classification frontend tests

### LINKED ISSUE #605: Add classification front-end tests

### PR DESCRIPTION
**Pull request by acrellin**
Closes #605

### CODE REVIEW

**Comment by acrellin** [comment]
[ch1087]

File: skyportal/tests/frontend/test_sources.py [hunk 1]
```diff
@@ -55,29 +56,70 @@ def test_classifications(driver, user, taxonomy_token, public_group, public_sour
...
-    taxonomy_button = driver.wait_for_xpath(f'//div[@id="tax-select"]')
-    taxonomy_button.click()
+    driver.wait_for_xpath_to_be_clickable('//div[@id="tax-select"]').click()
+    driver.wait_for_xpath_to_be_clickable(
+        f'//*[text()="{tax_name} ({tax_version})"]'
+    ).click()
+    driver.wait_for_xpath("//body").click()
+    class_input = driver.wait_for_xpath('//*[@id="classification"]')
+    driver.scroll_to_element_and_click(class_input)
+    class_input.send_keys("Symmetrical", Keys.ENTER)
```
Comment (profjsb) [inline skyportal/tests/frontend/test_sources.py:85]: `Symmetrical` isn't apart of the `simple` taxonomy (it is apart of `tdtax`). I'm surprised this is working.

File: skyportal/tests/frontend/test_sources.py [hunk 2]
```diff
@@ -55,29 +56,70 @@ def test_classifications(driver, user, taxonomy_token, public_group, public_sour
...
+    driver.get(f"/become_user/{user.id}")
     driver.get(f"/source/{public_source.id}")
     driver.wait_for_xpath(f'//div[text()="{public_source.id}"]')
-    taxonomy_button = driver.wait_for_xpath(f'//div[@id="tax-select"]')
-    taxonomy_button.click()
+    driver.wait_for_xpath_to_be_clickable('//div[@id="tax-select"]').click()
+    driver.wait_for_xpath_to_be_clickable(
+        f'//*[text()="{tax_name} ({tax_version})"]'
+    ).click()
+    driver.wait_for_xpath("//body").click()
```
Comment (profjsb) [inline skyportal/tests/frontend/test_sources.py:82]: why isn't the wait for `body` happening sooner? Like before `'//div[@id="tax-select"]'`?

File: skyportal/tests/frontend/test_sources.py [hunk 3]
```diff
@@ -55,29 +56,70 @@ def test_classifications(driver, user, taxonomy_token, public_group, public_sour
...
+    # Classifications list entry
+    driver.wait_for_xpath(f"//i[text()='{tax_name}']")
+    prob_span = driver.wait_for_xpath("//*[contains(text(), '(P=1)')]")
+    classification_entry_div = prob_span.find_element_by_xpath(
+        ".."
+    ).find_element_by_xpath("..")
+    # Hover to raise delete button
+    ActionChains(driver).move_to_element(classification_entry_div).perform()
+    delete_button = driver.wait_for_xpath(
+        "//button[starts-with(@name, 'deleteClassificationButton')]"
```
Comment (profjsb) [inline skyportal/tests/frontend/test_sources.py:104]: What happens if there are more than 1 classifications (like from a previous test run)? I think this XPATH is not specific enough to action on the specific delete button and may delete some other classification.

File: skyportal/tests/frontend/test_sources.py (hunk 1 above)
Comment (acrellin) [reply skyportal/tests/frontend/test_sources.py:85]: @profjsb 
``` python 
    simple = {
        'class': 'Cepheid',
        'tags': ['giant/supergiant', 'instability strip', 'standard candle'],
        'other names': ['Cep', 'CEP'],
        'subclasses': [
            {'class': 'Anomolous', 'other names': ['Anomolous Cepheid', 'BLBOO']},
            {
                'class': 'Mult-mode',
                'other names': ['Double-mode Cepheid', 'Multi-mode Cepheid', 'CEP(B)'],
            },
            {
                'class': 'Classical',
                'tags': [],
                'other names': [
                    'Population I Cepheid',
                    'Type I Cepheid',
                    'DCEP',
                    'Delta Cepheid',
                    'Classical Cepheid',
                ],
                'subclasses': [
                    {
                        'class': 'Symmetrical',
                        'other names': ['DCEPS', 'Delta Cep-type Symmetrical'],
                    }
                ],
            },
        ],
    }
```
I believe you wrote this.

File: skyportal/tests/frontend/test_sources.py (hunk 2 above)
Comment (acrellin) [reply skyportal/tests/frontend/test_sources.py:82]: @profjsb this is just to fully close the MUI select.

File: skyportal/tests/frontend/test_sources.py (hunk 3 above)
Comment (acrellin) [reply skyportal/tests/frontend/test_sources.py]: @profjsb a new test source is created on each run, so any old classifications will be associated with a different source, not the one being used each time the test is run again

File: skyportal/tests/frontend/test_sources.py (hunk 2 above)
Comment (acrellin) [reply skyportal/tests/frontend/test_sources.py:82]: See, e.g., `skyportal/tests/frontend/test_upload_photometry.py`

**Comment by acrellin** [comment]
> So beautiful I can cry.

:joy:

**Pull request merged**